backend/planty/dashboard/views.py

```diff
@@ -290,3 +295,234 @@ def post(self, request: Request):
+                'insolation': instruction.insolation,
+                'fertilizing': instruction.fertilizing,
+            }
+            my_instructions_json.append(instruction_json)
+
+
+        if request.method == 'GET' and 'limit_popular' in request.GET:
+            N = request.GET['limit_popular']
+        else:
+            N = 10
+
+        suggested = Instruction.objects.filter(public='True')
+        # returns best N suggested instruction
+        desc_suggested_instruction = suggested.order_by('num_selected').reverse()[:N]
+
+
+        desc_suggested_instructions_json = []
+        for instruction in desc_suggested_instruction:
+            instruction_json = {
+                'id': str(instruction.id),
+                'name': instruction.name,
+                'species': instruction.species,
+
+                'watering': instruction.watering,
+                'insolation': instruction.insolation,
+                'fertilizing': instruction.fertilizing,
+
+                'num_selected': instruction.num_selected
+            }
+            desc_suggested_instruction.append(instruction_json)
+
+        instructions_json = {
+            my: my_instructions_json,
+            suggested: desc_suggested_instructions_json
+        }
+
+        return Response(instructions_json, status=status.HTTP_200_OK)
+
+    def post(self, request: Request):
+        user: User = request.user
+        serializer = InstructionCreateSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        data = serializer.validated_data
+
+        instruction: Instruction = Instruction.objects.create(
+            id=uuid4(),
+            name=name,
+
+            user=user,
+            species=data['species'],
+
+            watering=data['watering'],
+            insolation=data['insolation'],
+            fertilizing=data['fertilizing']
+        )
+        instruction.save()
+
+        return Response(status=status.HTTP_201_CREATED)
+
+    def put(self, request: Request):
+        user: User = request.user
+        serializer = InstructionSelectSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        data = serializer.validated_data
+
+        try:
+            instruction: Instruction = Instruction.objects.get(pk=data['id'])
+        except Model.DoesNotExist:
+            return Response(data={
+                'id': ['Plant with the given ID does not exist']
+            }, status=status.HTTP_404_NOT_FOUND)
+
+        if instruction.user != user:
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        if instruction.public == 'True':
+            return Response(data={
+                'id:': ['Instruction with this ID is public. Cannot modify public instructions.']
+            }, status=status.HTTP_404_NOT_FOUND)
```

Comment:
Jeśli instrukcja istnieje, ale nie można jej zmodyfikować to request na pewno nie powinien zwracać 404.
Rozważyłbym 405.

Reply:
zmienione na `status=status.HTTP_405_METHOD_NOT_ALLOWED`